From c6b37c23621318b55b8293238f7f18cf465d555a Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 16 Jan 2026 14:00:48 +1100
Subject: [PATCH 5/5] reflect name of cuda driver library on windows

python effectively depends on the SystemRoot env var already anyway:
https://github.com/python/cpython/issues/118234
---
 tinygrad/runtime/autogen/cuda.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tinygrad/runtime/autogen/cuda.py b/tinygrad/runtime/autogen/cuda.py
index 55c101aec..967f9e9d1 100644
--- a/tinygrad/runtime/autogen/cuda.py
+++ b/tinygrad/runtime/autogen/cuda.py
@@ -7,6 +7,7 @@
 # LONGDOUBLE_SIZE is: 16
 #
 import ctypes, ctypes.util
+import os
 
 
 class AsDictMixin:
@@ -145,7 +146,8 @@ def char_pointer_cast(string, encoding='utf-8'):
 
 
 _libraries = {}
-_libraries['libcuda.so'] = ctypes.CDLL(ctypes.util.find_library('cuda'))
+with os.add_dll_directory(os.path.expandvars(r"%SystemRoot%\System32")):
+    _libraries['libcuda.so'] = ctypes.CDLL("nvcuda.dll")
 
 
 cuuint32_t = ctypes.c_uint32
